modules/tutorial-advanced-minimalist-fulcro/pages/index.adoc

```diff
@@ -0,0 +1,338 @@
+
+==== On router timeouts and errors
+
+What happens when loading data in a deferred route takes too long? What if it fails? The router has you covered:
+
+.Router with timeouts, a loading indicator, and an error message
+====
+```clojure
+(dr/defrouter MyRouter [_ {:keys [current-state route-factory route-props]}]
+  {:router-targets [AllPeople Person]}
+  ;; The body of the router is displayed only when the target is not ready,
+  ;; i.e. in one of the states below (unless you set `:always-render-body?`)
+  (case current-state ; <1>
+    nil (println "MISTAKE: MyRouter is displayed but has never been routed to yet") ; <2>
+    :pending (dom/div "Loading...") ; <3>
+    :failed (dom/div "Failed!")     ; <4>
+    (println "Should never come here:" current-state))) ; <5>
+```
+<1> The body of the router is only displayed when the target is not ready (unless you set `:always-render-body? true`). Thus the only router states we will ever see here are those listed below.
+<2> `current-state = nil` is the initial state, before the router has been routed to - i.e. when you failed to initialize it properly via `change-route!`, as described above (there is also the `:inital` state but I have not seen it in practice)
+<3> `:pending` is active for a deferred route, after the `:deferred-timeout` passed but before the `:error-timeout` - it also ends if `dr/target-ready` is transacted for the target ident
+<4> `:failed` if the deferred route has not become ready (as indicated by `dr/target-ready`) within the `:error-timeout` since the route event
+<5> We do not expect to see any other states, namely `:routed` - because in this state the target itself is displayed and not the router's body (unless we set `:always-render-body? true`)
+====
+
+Key points to know:
+
+* There are two timeouts that control the router's behavior with respect to deferred routes, `:deferred-timeout` (default 20ms)
+  and `:error-timeout` (default 5s). You can override the default values when calling `dr/change-route!`, link:{url-cljdoc-fulcro}com.fulcrologic.fulcro.routing.dynamic-routing#change-route![see its docstring].
+* You can use the body of the router and its `current-state`. It is only displayed when the target is not "ready" to be displayed
+* The router switches from displaying the previous target to displaying the router's body while waiting for a pending target only after the `:deferred-timeout` has passed (but it switches to the target immediately when it gets ready)
+* If the mutation `(dr/target-ready {:target <ident>}` is not triggered within `:error-timeout` of `change-route!` then the state becomes `:failed` (the target can still become "ready" later, if the load eventually finishes, and the target will be displayed)
+* Thus we can use `:pending` to display a loading indicator, if it takes longer than the `:deferred-timeout`. We can use the `:failed` state to show an error message telling the user it is either taking too long or failed
+
+=== Nesting routers
+
```

Comment:
Hukka: Another question I came up is does fulcro routing support multiple parallel, not nested, route trees. 

It does but I felt it is out of scope of the tutorial. But I guess I could/should point to the Book on this.